modules/phase_field/include/materials/OPInterfaceBarrierMaterial.h

```diff
@@ -0,0 +1,49 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef OPINTERFACEBARRIER_H
+#define OPINTERFACEBARRIER_H
+
+#include "Material.h"
+#include "DerivativeMaterialInterface.h"
+
+// Forward Declarations
+class OPInterfaceBarrierMaterial;
+
+template<>
+InputParameters validParams<OPInterfaceBarrierMaterial>();
+
+/**
+ * OPInterfaceBarrierMaterial is a constraint kernel that acts on all of the eta_i variables to
+ * prevent more than two eta variables going above 0 on an interface.
+ */
+class OPInterfaceBarrierMaterial : public DerivativeMaterialInterface<Material>
+{
+public:
+  OPInterfaceBarrierMaterial(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpProperties();
+
+  /// name of the function of eta (used to generate the material property names)
+  std::string _function_name;
+
+  /// Polynomial order of the barrier function \f$ g(\eta) \f$
+  MooseEnum _g_order;
+
+  /// order parameters
+  unsigned int _num_eta;
+  std::vector<VariableValue *> _eta;
```

Comment:
This line needs to hold constant pointers due to a change we made in the framework recently.

```
std::vector<const VariableValue *> _eta;
```